internal/dataplane/parser/translate_tlsroute.go

```diff
@@ -75,3 +91,52 @@ func (p *Parser) ingressRulesFromTLSRoute(result *ingressRules, tlsroute *gatewa
 
 	return nil
 }
+
+// isTLSRoutePassthrough returns true if we need to configure TLS passthrough to kong
+// for the tlsroute object.
+// returns a non-nil error if we failed to get the supported gateway.
+func (p *Parser) isTLSRoutePassthrough(tlsroute *gatewayv1alpha2.TLSRoute) (bool, error) {
+	// reconcile loop will push TLSRoute object with updated status when
+	// gateway is ready and TLSRoute object becomes stable.
+	// so we get the supported gateways from status.parents.
+	for _, parentStatus := range tlsroute.Status.Parents {
+		parentRef := parentStatus.ParentRef
+
+		if parentRef.Group != nil && string(*parentRef.Group) != gatewayv1beta1.GroupName {
+			continue
+		}
+
+		if parentRef.Kind != nil && gatewayv1beta1.Kind(*parentRef.Kind) != KindGateway {
+			continue
+		}
+
+		gatewayNamespace := tlsroute.Namespace
+		if parentRef.Namespace != nil {
+			gatewayNamespace = string(*parentRef.Namespace)
+		}
+
+		gateway, err := p.storer.GetGateway(gatewayNamespace, string(parentRef.Name))
+		if err != nil {
+			if errors.As(err, &store.ErrNotFound{}) {
+				continue
```

Comment:
Should we at least log here? This seems like it would be a pretty unusual situation--I don't think we should be able to end up with a TLSRoute that has no parent Gateway (else we wouldn't be processing it) and the status Gateway should be consistent with the store Gateway, since they're both set by the controller.

Reply:
added logs here.